tests/rust/aligned_struct.toml

```diff
@@ -0,0 +1,21 @@
+[struct]
+packed = "__attribute__ ((packed))"
+aligned = "__attribute__ ((aligned(<...>)))"
```

Comment:
I don't think this is very discoverable... I think we should just take a macro name and assume it's functional, so like:

```
aligned = "CBINDGEN_ALIGNED"
```

And then have in the `header`:

```
#define CBINDGEN_ALIGNED(n) __attribute__ ((aligned(n)))
```

WDYT?

Reply:
Sure, as long as it's documented. However, it's unclear to me how we should handle `#[repr(packed(x))]` -- should there be two macros or just one that always takes an argument (and for `#[repr(packed)]` we pass an alignment of `1`)? In C, I'm pretty sure the equivalent of `#[repr(packed(x))]` is `__attribute__((packed, aligned(x)))` which is a problem because `#[repr(packed)]` is `__attribute__((packed))` not `__attribute__((packed, aligned(1)))`. So AFAICS we'd need two different macros?

Reply:
Should we have any of those settings by default? Or should we keep it purely optional like `must_use`? Arguably it's not really correct to generate a non-opaque version of a structure without including it's alignment (and *especially* if we don't include it's packing).

Reply:
I think what you did (dropping if we detect we'd generate bad code) is the right thing to do.